src/main/java/com/onionshop/controllers/ProjectStateController.java

```diff
@@ -248,4 +263,23 @@ public void onCanvasMouseReleased(MouseEvent mouseDragEvent) {
         }
         projectManager.updateDrawingCanvas(pixelArray);
     }
+
+    /**
+     * This function checks if the current tool is a shape and if it is it draws a shape on the canvas when the
+     * user releases their mouse
+     *
+     * @param mouseDragEvent The MouseEvent triggered when the user releases their mouse
+     */
+    public void updateShapeOnMouseRelease(MouseEvent mouseDragEvent) {
+        // These two if statement check if the current tool as a shape that needs to be drawn with click and drag
+        if (ToolStateManager.getInstance().getCurrentToolState() instanceof Shape) {
```

Comment:
I mean you could add like a getShape method to toolStateManager which returns the shape if it is a shape, but this is also fine. i dont think it matters too much